packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1133,23 +1139,26 @@ export class ProjectLifecycleManager {
       return metaState
     }
 
-    const configFileTs = this._pathToFile('cypress.config.ts')
-    const configFileJs = this._pathToFile('cypress.config.js')
+    for (const fileName of potentialConfigFiles) {
+      const filePath = this._pathToFile(fileName)
+      const fileExists = fs.existsSync(filePath)
 
-    if (fs.existsSync(configFileTs)) {
-      metaState.hasValidConfigFile = true
-      this.setConfigFilePath('cypress.config.ts')
-    }
+      if (fileExists) {
+        // We'll collect all the found config files.
+        // If we found more than one, this list will be used in an error message.
+        metaState.allFoundConfigFiles.push(fileName)
```

Comment:
This is mutating the `allFoundConfigFiles` array globally, we probably need to create one above where we do:

```diff
const metaState: ProjectMetaState = {
  ...PROJECT_META_STATE,
  hasLegacyCypressJson: fs.existsSync(this._pathToFile(this.legacyConfigFile)),
  hasCypressEnvFile: fs.existsSync(this._pathToFile('cypress.env.json')),
+ allFoundConfigFiles: [],  
}
```

Reply:
I think we can type this to avoid the problem with this one neat trick

```ts
export interface SetupFullConfigOptions {
  allFoundConfigFiles: readonly string[]
}
```

![image](https://user-images.githubusercontent.com/19196536/157786868-1b934151-a86e-4523-a96f-9bc55a6de854.png)

Reply:
I see @tgriesser. I ended up resetting it right above where I push onto it.

Reply:
Can we type it like above @JessicaSachs ?

Reply:
We actually want to push into it though.